### DISCUSSION TITLE
Websocket in specific Nextjs pages

### DISCUSSION THREAD

**Discussion by ibrahim-bournine** [trpc/trpc#2672] [Q&A] [answered]
So i've been trying to get websockets to work on specific pages and it doesn't seem possible, using the **splitLink** function still establishes a ws connection. and applying the HOF **withTrpc** on each page breaks ssr for some reason.

it seems resource inefficient to open a ws connection in every page, especially in my case where the vast majority of users won't end up in  a page that has a useSubscription hook.

i'm using trpc v9 btw.

**Comment by KATT** [comment, marked as answer]
I've been considering adding a `lazy` mode on connections that could establish the WS connection only when it's used and automatically disconnect (with a grace period?) thereafter. Would that solve your issue, you think?

**Reply by ibrahim-bournine** [reply]
yes, that would be perfect

**Reply by KATT** [reply]
Cool. Feel free to build it and submit a PR, I don't use websockets myself at the moment so I won't build it anytime soon. Can also agree to do it if you convince your company to join one of the top 3 sponsorship tiers ☺

**Reply by ibrahim-bournine** [reply]
@KATT  i just submitted a PR #2765.
